astroquery/mast/observations.py

```diff
@@ -740,12 +740,12 @@ def download_products(self, products, *, download_dir=None, flat=False,
 
             products = vstack(product_lists)
 
-        # Remove duplicate products
-        products = self._remove_duplicate_products(products)
-
         # apply filters
         products = self.filter_products(products, mrp_only=mrp_only, **filters)
 
+        # remove duplicate products
+        products = self._remove_duplicate_products(products)
```

Comment:
Would this cause some performance degradation as now the duplicates end up going into `filter_products()`? Does filtering or duplicate removal is more costly, or it doesn't matter in practice?

Reply:
Hmm, I guess it would depend on the particular query. You may have a good amount of duplicate products, but if you set strict filters (like `mrp_only = True` in some cases), it would be better to filter first. The time complexity of either filtering or removing duplicates would be O(n). I don't think that it matters too much in practice.

Reply:
OK, then I will leave this to your judgment as the maintainer of this module, and merge in as is.

Thanks!